From 1058d68fff272f9399542acf5c982c2bc95c4eaf Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Wed, 7 Mar 2018 22:49:10 +0200
Subject: [PATCH 1037/5242] MLK-12746-1: ARM: dts: add QSPI support on
 i.MX6ULL

commit  816af93dc5236743b5d9636d34392bf1adfbef77 from
https://source.codeaurora.org/external/imx/linux-imx.git

support QSPI on i.MX6ULL. By default, only QSPI1 was enabled, while
reworked board could support all 4 QSPI chips.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                         |    1 +
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-qspi.dts |   18 ++++++++++++++++++
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts      |    7 ++++++-
 3 files changed, 25 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-qspi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 3506171..47db6a4 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -603,6 +603,7 @@ dtb-$(CONFIG_SOC_IMX6ULL) += \
 	imx6ull-14x14-ddr3-arm2-ecspi.dtb \
 	imx6ull-14x14-ddr3-arm2-flexcan2.dtb \
 	imx6ull-14x14-ddr3-arm2-lcdif.dtb \
+	imx6ull-14x14-ddr3-arm2-qspi.dtb \
 	imx6ull-14x14-ddr3-arm2-gpmi-weim.dtb \
 	imx6ull-14x14-ddr3-arm2-uart2.dtb \
 	imx6ull-14x14-ddr3-arm2-usb.dtb \
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-qspi.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-qspi.dts
new file mode 100644
index 0000000..63659bd
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-qspi.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#define REWORKED_ENABLE_ALL_QSPI
+#include "imx6ull-14x14-ddr3-arm2.dts"
+
+&gpmi {
+	status = "disabled";
+};
+
+&usdhc2 {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
index 8486b14..3642016 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2.dts
@@ -905,8 +905,10 @@
 &qspi {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_qspi>;
-	status = "disabled";
+	status = "okay";
+#ifdef REWORKED_ENABLE_ALL_QSPI
 	fsl,qspi-has-second-chip = <1>;
+#endif
 	ddrsmp=<0>;
 
 	flash0: n25q256a@0 {
@@ -918,6 +920,8 @@
 		reg = <0>;
 	};
 
+#ifdef REWORKED_ENABLE_ALL_QSPI
+
 	flash1: n25q256a@1 {
 		#address-cells = <1>;
 		#size-cells = <1>;
@@ -944,6 +948,7 @@
 		spi-nor,ddr-quad-read-dummy = <6>;
 		reg = <3>;
 	};
+#endif
 };
 
 &uart1 {
-- 
1.7.9.5

